plugins/modules/sysctl.py

```diff
@@ -187,6 +213,16 @@ def process(self):
 
         # Do the work
         if not self.module.check_mode:
```

Comment:
There is some design issue here. `check_mode` should show what would be done, without actually doing it, that doesn't seem to be the case here. As said by its name, `module.set_fs_attributes_if_different()` set attributes if different. It handles `check_mode` on its own and consistently (as well as `changed` status and `diff`).